test/08-tasks.test.ts

```diff
@@ -408,5 +409,66 @@ describe('08 task tests', function () {
 				})
 				.expect(500);
 		});
+
+		it('should not pick up tasks while worker is not running', async () => {
+			const name = randomUUID();
+			const task = await createTask(pineTest, apikey, {
+				is_executed_by__handler: 'create_device',
+				is_executed_with__parameter_set: {
+					name,
+					type: randomUUID(),
+				},
+				is_scheduled_to_execute_on__time: new Date(Date.now() + 2000),
+			});
+
+			pineServer.send(PINE_TEST_SIGNALS.STOP_TASK_WORKER);
+
+			// Wait a few seconds to ensure the task is not executed
+			await setTimeout(3000);
```

Comment:
You could also stop the worker first and then create the task. This makes the `is_scheduled_to_execute_on__time` unnecessary and safe to reduce the wait from 3s to say 2s. Removing the future execution argument also makes the test clearer as it removes any ambiguity as to why the task may have not been executed (it's definitely because the worker is stopped). What do you think?

Reply:
I think it is a great idea, if we stop the worker first and then create the task, I believe we don't need any `await setTimeout` in fact. There is maybe a very very tiny chance of the task being picked up while the event loop process the stop promise, but I think it is so small that we wouldn't see any intermitency ever. I ran the test locally 1k times and it succeded in all of them without any await.